histograms/histogram.go

```diff
@@ -0,0 +1,83 @@
+package histograms
+
+import (
+	"sync"
+	"time"
+
+	"github.com/wavefronthq/wavefront-sdk-go/senders"
+
+	tdigest "github.com/caio/go-tdigest"
+)
+
+type Histogram struct {
```

Comment:
We should define an interface named `Histogram` with the relevant exported methods and have this struct just implement that type. Can rename this struct to something else.